src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -334,25 +382,35 @@ class RegularSync(
       val blocks = headersQueue.zip(m).map{ case (header, body) => Block(header, body) }
 
       @tailrec
-      def importBlocks(blocks: List[Block], importedBlocks: List[Block] = Nil): (List[Block], Option[BlockImportResult]) =
+      def importBlocks(blocks: List[Block], importedBlocks: List[Block] = Nil): (List[Block], Option[Any]) =
```

Comment:
Cant we create a `BlockImportResult` for the missing MPT node to avoid `Any`?

Reply:
I'd rather not do that. All errors extending `BlockImportResult` are logical errors, that are relevant to the nature of the blockchain. `MissingNodeException` is a result of our implementation and in the perfect world shouldn't even be handled.

I agree that `Option[Any]` does look a bit clumsy, but I would not be overly concerned with that as it's a nested function (well, it will be private now, but still).